src/core/utils/compatibility.js

```diff
@@ -160,3 +162,17 @@ export function getClientCompatibility({
     reason,
   };
 }
+
+export const isQuantumCompatible = ({ addon }) => {
+  // HACK: we use a hardcoded User Agent string corresponding to MacOS/Firefox
+  // 57.0 to determine whether the given `addon` is compatible with Quantum.
+  // The OS is not important as far as it is not `iOS`.
+  //
+  // See: https://github.com/mozilla/addons-frontend/issues/3868.
+  const userAgentInfo = UAParser(oneLine`Mozilla/5.0 (Macintosh; Intel Mac OS X
+    10.12; rv:57.0) Gecko/20100101 Firefox/57.0`);
+  // We set `clientApp` to `firefox` because we do not want to rely on it.
+  const clientApp = CLIENT_APP_FIREFOX;
```

Comment:
Actually, I think we do need to rely on `clientApp` because that's crucial to how the question "is this add-on compatible with 57" is answered. If an add-on is compatible with Firefox for Android 57 but is *not* compatible with Firefox for desktop 57 then we *do not* want to show the "incompatible with Quantum" banner.